scraper_common.js

```diff
@@ -0,0 +1,167 @@
+                .catch((error) => {
+                    //print out the issue but don't fail, this way we still publish updates
+                    //for other locations even if this website's scrape doesn't work
+                    console.log(error);
+                    isSuccess = false;
+                    return null;
+                })
+                .then(async (result) => {
+                    const numberAppointments = getTotalNumberOfAppointments(
+                        result
+                    );
+                    // TODO - call FaunaDB util method here!
+                    await logScraperRun(
+                        scraper.name,
+                        isSuccess,
+                        new Date() - startTime,
+                        startTime,
+                        numberAppointments
+                    );
+                    return result;
+                });
+            results.push(returnValue);
+        }
+        if (usePuppeteer) {
+            await browser.close();
+        }
+        let scrapedResultsArray = [];
+        for (const result of results) {
+            if (Array.isArray(result)) {
+                scrapedResultsArray.push(...result);
+            } else if (result) {
+                //ignore nulls
+                scrapedResultsArray.push(result);
+            }
+        }
+
+        const cachedResults = await fetch(
+            "https://mzqsa4noec.execute-api.us-east-1.amazonaws.com/prod"
+        )
+            .then((res) => res.json())
+            .then((unpack) => JSON.parse(unpack.body).results);
+
+        let finalResultsArray = [];
+        if (process.argv.length <= 2) {
+            // Only add default data if we're not testing individual scrapers.
+            // We are not passing in the optional 3rd arg of mergeResults;
+            // this means that there is no time limit on stale data being merged in.
+            finalResultsArray = dataDefaulter.mergeResults(
+                scrapedResultsArray,
+                cachedResults
+            );
+        } else {
+            finalResultsArray = scrapedResultsArray;
+        }
+
+        const responseJson = {
+            // Version number of the file
+            version: 1,
+
+            // Timestamp for the archived data.json file.
+            timestamp: s3.getTimestampForFile(),
+
+            // Add geocoding for all locations
+            results: await getAllCoordinates(finalResultsArray, cachedResults),
+        };
+
+        const webData = JSON.stringify(responseJson);
+
+        if (process.env.DEVELOPMENT) {
+            const outFile = usePuppeteer ? "out.json" : "out_no_browser.json";
+            console.log(
+                "The data that would be published is in '" + outFile + "'"
+            );
+            //console.log("The following data would be published:");
+            //console.dir(responseJson, { depth: null });
+            file.write(outFile, webData);
+            /*
```

Comment:
here too?

Reply:
@livgust commented those out in a prior release.  I added the new console.log that tells the developer where the file is.